### PR TITLE
httpd: rebuild to fix config vars

### PR DESCRIPTION
**Pull request by shivammathur**
- [X] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [X] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [X] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [X] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

Currently, `apxs` is not giving the correct path for the config variable `APU_CONFIG`.

```zsh
$ apxs -q APU_CONFIG       
/opt/homebrew/opt/apr-util/libexec/bin/apu-1-config
apxs:Error: /opt/homebrew/opt/apr-util/libexec/bin/apu-1-config not found!.
```

This breaks building PHP
https://github.com/shivammathur/homebrew-php/runs/6204315806?check_suite_focus=true#step:8:504

Reinstalling `httpd` from the source fixes the issue.
```zsh
$ brew reinstall -s httpd
...
$ apxs -q APU_CONFIG
/opt/homebrew/opt/apr-util/bin/apu-1-config
```

### CODE REVIEW

**Comment by carlocab** [comment]
Can we add a test that catches this so that it gets rebuilt ahead of time next time around?

**Comment by shivammathur** [comment]
@carlocab 

Add a test to output the `APU_CONFIG`. It will fail if the config is incorrect.

File: Formula/httpd.rb [hunk 1]
```diff
@@ -171,6 +171,8 @@ def caveats
       sleep 3
 
       assert_match expected_output, shell_output("curl -s 127.0.0.1:#{port}")
+
+      exec bin/"apxs", "-q", "APU_CONFIG"
```
Comment (carlocab) [inline Formula/httpd.rb:175]:
```suggestion
      # Check that `apxs` can find `apu-1-config`.
      system bin/"apxs", "-q", "APU_CONFIG"
```
I don't think `exec` will error on a non-zero exit code. Let's also add a comment so what's happening here is clearer.

Comment (shivammathur) [reply Formula/httpd.rb]: Done

**Comment by carlocab** [comment]
Only the ephemeral ARM runner failed. Doesn't seem to be a `python2` issue. Odd. @Bo98, thoughts?

**Review by carlocab — DISMISSED** [review]

**Comment by SMillerDev** [comment]
Should probably add a dependency on python then since 12.3 does not have v2 anymore

**Comment by carlocab** [comment]
Does this need Python? The error doesn't seem to be Python-related, but maybe I'm reading this wrong.

**Comment by SMillerDev** [comment]
No, I read your message wrong. Does**n't**, seem to be python.

**Comment by Bo98** [comment]
Thanks for the ping. I've actually seen a user report the same build error, so it does seem like a genuine issue.

I don't quite understand why it doesn't happen on the other runners however.

**Comment by Bo98** [comment]
Only thing I can think is that `ENV.deparallelize` might need to be there for all platforms - not just Linux? Or it might not make a difference, not sure.

> ```
> # suexec does not install without root
> bin.install "support/suexec"
> ```

This seems incorrect given the script would fail if it doesn't install?

Can add a test to check it exists if we want to be sure though.

> `--sbindir=#{bin}`

Why?

**Comment by carlocab** [comment]
> > `--sbindir=#{bin}`
> 
> 
> 
> Why?

Not sure. I've seen this in a number of very old formulae. My guess is that `brew` used to not link `sbin` directories so we moved stuff to `bin` instead.

**Comment by SMillerDev** [comment]
```
  chmod: Unable to change file mode on /opt/homebrew/Cellar/httpd/2.4.53_2/bin/suexec: Operation not permitted
  make[1]: *** [install-suexec-setuid] Error 1
  make: *** [install-recursive] Error 1
```

**Comment by Bo98** [comment]
Not sure if it's best to ship the sbin change or not for backwards compatibility reasons. I'm just trying to narrow down the issue.

**Comment by shivammathur** [comment]
@Bo98 
I agree, that shipping the sbin change will break backward compatibility.
If we can skip the ephemeral build and merge this PR, that would be great.

**Comment by shivammathur** [comment]
@Bo98 

> Rebase the branch back to [aca9dc1](https://github.com/Homebrew/homebrew-core/commit/aca9dc103dfea78e1d178f3c33efc315a58eaa5d) and I'll get this merged.

Done

**Review by Bo98 — APPROVED** [review]

**Pull request closed without merging**